raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -132,8 +133,13 @@ public long timeUntilCheckQuorumExpires(long currentTimeMs) {
      */
     public void updateCheckQuorumForFollowingVoter(int id, long currentTimeMs) {
         updateFetchedVoters(id);
-        // The majority number of the voters excluding the leader. Ex: 3 voters, the value will be 1
-        int majority = voterStates.size() / 2;
+        // The majority number of the voters. Ex: 2 for 3 voters, 3 for 4 voters... etc.
+        int majority = (int) Math.ceil((double) (voterStates.size() + 1) / 2);
+        // Check if the leader is removed from the voter set
+        if (voterStates.containsKey(localId)) {
+            majority = majority - 1;
```

Comment:
Let's add a comment explaining why we are doing this. This is my understanding:
> If the leader is in the voter set since it should be implicitly counted as part of the majority but the leader will never be a member of the `fetchedVoters`. If the leader is not in the voter set, it is not in majority so the majority can only be composed of fetched voters.

Do you agree?

Reply:
Looks good to me. Thanks.